src/publishProject/projectUtils.ts

```diff
@@ -0,0 +1,172 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as mssql from "vscode-mssql";
+import * as constants from "../constants/constants";
+import { SqlProjectsService } from "../services/sqlProjectsService";
+
+// Shape returned by sqlProjectsService.getProjectProperties (partial, only fields we use)
+export interface ProjectProperties {
+    projectGuid?: string;
+    configuration?: string;
+    outputPath: string;
+    databaseSource?: string;
+    defaultCollation: string;
+    databaseSchemaProvider: string;
+    projectStyle: unknown;
+    targetVersion?: string;
+}
+
+/**
+ * Target platforms for a sql project
+ */
+export const enum SqlTargetPlatform {
+    sqlServer2012 = "SQL Server 2012",
+    sqlServer2014 = "SQL Server 2014",
+    sqlServer2016 = "SQL Server 2016",
+    sqlServer2017 = "SQL Server 2017",
+    sqlServer2019 = "SQL Server 2019",
+    sqlServer2022 = "SQL Server 2022",
+    sqlAzure = "Azure SQL Database",
+    sqlDW = "Azure Synapse SQL Pool",
+    sqlEdge = "Azure SQL Edge",
```

Comment:
Can you double-check that we still should have Edge here?  I thought it was no longer supported.